Sources/MapLibreSwiftDSL/Style Layers/Symbol.swift

```diff
@@ -40,10 +53,9 @@ public struct SymbolStyleLayer: SourceBoundVectorStyleLayerDefinition {
         return SymbolStyleLayerInternal(definition: self, mglSource: styleSource)
     }
 
-    // TODO: Other properties and their modifiers
-    fileprivate var iconImageName: NSExpression?
+    public var iconImageName: NSExpression?
 
-    private var iconImages = [UIImage]()
+    public var iconImages = [UIImage]()
```

Comment:
I made these two public, because this allows devs to make their own extensions for the SymbolStyleLayer. For example, we created a helper extension for ourselves like this, but needed these properties exposed to do so:

```swift
public extension SymbolStyleLayer {

    func iconImage(mappings: [AnyHashable: UIImage], default defaultImage: UIImage) -> Self {
        return modified(self) { it in

            let expression1 = NSExpression(forKeyPath: "ios_category_icon_name")
            let expression2 = NSExpression(forKeyPath: "ios_category_icon_color")

            // Create an NSExpression that concatenates the two key paths
            let attributeExpression = expression1.mgl_appending(expression2)
            let mappingExpressions = mappings.mapValues { image in
                NSExpression(forConstantValue: image.sha256())
            }
            let mappingDictionary = NSDictionary(dictionary: mappingExpressions)
            let defaultExpression = NSExpression(forConstantValue: defaultImage.sha256())

            // swiftlint:disable force_cast
            it.iconImageName = NSExpression(
                forMLNMatchingKey: attributeExpression,
                in: mappingDictionary as! [NSExpression: NSExpression],
                default: defaultExpression
            )
            // swiftlint:enable force_cast
            it.iconImages = mappings.values + [defaultImage]
        }
    }

}
```

Reply:
That makes sense. Thanks for the detailed example clarifying the use case!